src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clas) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoClassAnnotation {
+        Constructor constructor = clas.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clas.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                BetDao betDao = Factory.getBetDao();
+                if (field.getType().equals(BetDao.class)
+                        && betDao.getClass().getAnnotation(Dao.class) != null) {
+                    field.setAccessible(true);
+                    field.set(instance, betDao);
+                } else {
+                    UserDao userDao = Factory.getUserDao();
+                    if (field.getType().equals(UserDao.class)
+                            && userDao.getClass().getAnnotation(Dao.class) != null) {
+                        field.setAccessible(true);
+                        field.set(instance, userDao);
+                    } else {
+                        throw new NoClassAnnotation("DaoImpl class has no @Dao annotation");
+                    }
+                }
```

Comment:
I would recommend to simplify it and change the name of your custom Exception

```suggestion
                BetDao betDao = Factory.getBetDao();
                UserDao userDao = Factory.getUserDao();
                field.setAccessible(true);
                if (field.getType().equals(BetDao.class)
                        && betDao.getClass().getAnnotation(Dao.class) != null) {
                    field.set(instance, betDao);
                } else if (field.getType().equals(UserDao.class)
                            && userDao.getClass().getAnnotation(Dao.class) != null) {
                    field.set(instance, userDao);
                } else {
                        throw new NoImplementationException(field.getName()
                            + " doesn't have an implementation that can be injected");
                }
```

Reply:
done